### ISSUE TITLE
Ongoing talk exits when the camera source is changed on call.

### ISSUE THREAD

**Issue by shashank-brdj** [nextcloud/spreed#6780]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Join a talk call using an external camera
2. talk starts with camera disabled. Try changing the camera source in options
3. Talk kicks you out into another empty talk lobby and you need to wait for other person to join back

### Expected behaviour
Tell us what should happen
Talk should work as intended, ie it should let the user change camera source as needed

### Actual behaviour
Tell us what happens instead
Talk crashes into a different lobby and kicks the other person out of the lobby when camera source is changed

## Talk app
13.0.1
**Talk app version:** (see apps admin page: `/index.php/settings/apps`)

**Custom Signaling server configured:** yes/no and version (see additional admin settings: `/index.php/index.php/settings/admin/talk#signaling_server`)

**Custom TURN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/talk#turn_server`)

**Custom STUN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/talk#stun_server`)

## Browser

**Microphone available:** yes/no
yes
**Camera available:** yes/no
yes
**Operating system:** Windows/Ubuntu/...
Ubuntu
**Browser name:** Firefox/Chrome/...
Brave
**Browser version:** 85/96/...
97.0.4692.71

### Browser log

<details>
```
Insert your browser log here, this could for example include:
a) The javascript console log
b) The network log
c) ...
```

</details>

## Server configuration

**Operating system**: Ubuntu/RedHat/...
Ubuntu Server 20.04
**Web server:** Apache/Nginx
Apache
**Database:** MySQL/Maria/SQLite/PostgreSQL
MariaDB
**PHP version:** 7.4/8.0/8.1
7.4
**Nextcloud Version:** (see admin page)
23
**List of activated apps:**

<details>
```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your server installation folder
```

Enabled:
  - accessibility: 1.9.0
  - activity: 2.15.0
  - breezedark: 23.1.0
  - bruteforcesettings: 2.3.0
  - calendar: 3.0.4
  - circles: 23.0.0
  - cloud_federation_api: 1.6.0
  - comments: 1.13.0
  - contacts: 4.0.7
  - contactsinteraction: 1.4.0
  - dashboard: 7.3.0
  - dav: 1.21.0
  - deck: 1.6.0
  - drawio: 1.0.2
  - federatedfilesharing: 1.13.0
  - federation: 1.13.0
  - files: 1.18.0
  - files_pdfviewer: 2.4.0
  - files_rightclick: 1.2.0
  - files_sharing: 1.15.0
  - files_trashbin: 1.13.0
  - files_versions: 1.16.0
  - files_videoplayer: 1.12.0
  - firstrunwizard: 2.12.0
  - forms: 2.4.0
  - logreader: 2.8.0
  - lookup_server_connector: 1.11.0
  - mail: 1.11.5
  - notes: 4.3.0
  - notifications: 2.11.1
  - oauth2: 1.11.0
  - password_policy: 1.13.0
  - passwords: 2021.12.20
  - photos: 1.5.0
  - privacy: 1.7.0
  - provisioning_api: 1.13.0
  - recommendations: 1.2.0
  - richdocuments: 5.0.1
  - richdocumentscode_arm64: 21.11.4
  - serverinfo: 1.13.0
  - settings: 1.5.0
  - sharebymail: 1.13.0
  - spreed: 13.0.1
  - support: 1.6.0
  - survey_client: 1.11.0
  - systemtags: 1.13.0
  - tasks: 0.14.2
  - text: 3.4.0
  - theming: 1.14.0
  - twofactor_backupcodes: 1.12.0
  - updatenotification: 1.13.0
  - user_status: 1.3.1
  - viewer: 1.7.0
  - weather_status: 1.3.0
  - workflowengine: 2.5.0

Disabled:
  - admin_audit
  - encryption
  - files_external
  - nextcloud_announcements
  - user_ldap
 
</details>

**Nextcloud configuration:**

<details>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder
```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "mydomain.com",
            "192.168.0.110"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "23.0.0.10",
        "overwrite.cli.url": "https:\/\/mydomain.com",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance": false,
        "default_phone_region": "1974",
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "app_install_overwrite": [
            "documentserver_community",
            "mydomain.com"
        ],
        "theme": ""
    }
}

</details>

### Server log (data/nextcloud.log)
<details>

```
Insert your server log here
```
</details>

**Comment by nickvergessen** [comment]
> Talk crashes into a different lobby and kicks the other person out of the lobby when camera source is changed

It's just you being reconnected. The connection between other users are unaffected by your change.
Nevertheless we fixed a related issue recently: #6771  
So please retest once 13.0.2 is out.

**Comment by shashank-brdj** [comment]
Sorry for the late reply, updated to Talk version 13.0.2. The issue has been resolved.
Thank you @nickvergessen
